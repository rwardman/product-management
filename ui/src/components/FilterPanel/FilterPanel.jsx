import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";

import * as Styled from "./FilterPanel.styles";

const FilterPanel = ({ categories, filterOn }) => {
  const [activeFilters, setActiveFilters] = useState([]);

  useEffect(() => {
    filterOn(activeFilters);
  }, [filterOn, activeFilters]);

  const handleChange = (id) => {
    var list = [...activeFilters];
    if (!list.includes(id)) {
      list.push(id);
      setActiveFilters(list);
    } else {
      setActiveFilters([...activeFilters].filter((word) => word !== id));
    }
  };

  return (
    <Styled.Form>
      <Styled.Label>
        <Styled.FilterTitle>Filter</Styled.FilterTitle>
        {categories.map((category) => (
          <label key={category.id}>
            {category.name}
            <input
              name={category.name}
              type="checkbox"
              onClick={() => handleChange(category.id)}
            />
          </label>
        ))}
      </Styled.Label>
    </Styled.Form>
  );
};

export default FilterPanel;

FilterPanel.propTypes = {
  filterOn: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
};
