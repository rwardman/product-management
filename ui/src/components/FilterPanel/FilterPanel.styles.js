import styled from "styled-components";

const Form = styled.form`
  border: 1px solid #aaa;
  border-radius: 3px;
  padding: 5px;
`;

const Label = styled.label`
  display: flex;
  flex-direction: column;
`;

const FilterTitle = styled.span`
  font-size: 16px;
  margin-top: 5px;
  margin-bottom: 5px;
  font-weight: bold;
`;

export { Form, Label, FilterTitle };
