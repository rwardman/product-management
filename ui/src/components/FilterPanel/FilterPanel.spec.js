import renderer from "react-test-renderer";
import FilterPanel from "./FilterPanel";

const categories = [
  { id: "id1", name: "name1" },
  { id: "id2", name: "name2" },
];

describe("When displaying the filter panel component", () => {
  it("should be displayed correctly", () => {
    const tree = renderer
      .create(<FilterPanel categories={categories} filterOn={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
