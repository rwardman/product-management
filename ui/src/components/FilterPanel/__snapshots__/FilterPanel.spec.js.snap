// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When displaying the filter panel component should be displayed correctly 1`] = `
<form
  className="sc-bdfBwQ icbxSF"
>
  <label
    className="sc-gsTCUz qIFjy"
  >
    <span
      className="sc-dlfnbm ipJTGa"
    >
      Filter
    </span>
    <label>
      name1
      <input
        name="name1"
        onClick={[Function]}
        type="checkbox"
      />
    </label>
    <label>
      name2
      <input
        name="name2"
        onClick={[Function]}
        type="checkbox"
      />
    </label>
  </label>
</form>
`;
